### PR TITLE
fix: leaking UI context GetE2EICertificateUseCase (WPB-6648)

### PR DESCRIPTION
**Pull request by alexandreferris**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6648" title="WPB-6648" target="_blank">BugWPB-6648</a>  [Android] GetE2EICertificateUseCase is leaking UI context
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [X] contains a reference JIRA issue number like `SQPIT-764`
    - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Leaking Context from `GetE2EICertificateUseCase`

### Causes (Optional)

Context was being passed from UI

### Solutions

Inject `applicationContext` on ViewModel that is calling `GetE2EICertificateUseCase`

### CODE REVIEW

**Review by saleniuk — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [3158](https://10.10.124.134/job/AR-build-pipeline/3158/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.0(8904382)(8904382)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.0(8904382)(8904382)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt [hunk 1]
```diff
@@ -59,7 +60,8 @@ class FeatureFlagNotificationViewModel @Inject constructor(
     private val currentSessionFlow: CurrentSessionFlowUseCase,
     private val globalDataStore: GlobalDataStore,
     private val disableAppLockUseCase: DisableAppLockUseCase,
-    private val dispatcherProvider: DispatcherProvider
+    private val dispatcherProvider: DispatcherProvider,
+    @ApplicationContext private val applicationContext: Context
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt]: There is no need to inject it into the view model, we can directly inject it into the use case constructor this way there is zero chance of someone passing a UI context to the use case

**Review by MohamadJaara — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt (hunk 1 above)
Comment (alexandreferris) [reply app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt]: As we are creating the usecase in this ViewModel we will need to pass something when creating the usecase 🤔

**Review by MohamadJaara — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [3172](https://10.10.124.134/job/AR-build-pipeline/3172/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.0(8911541)(8911541)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.0(8911541)(8911541)-staging-compat.apk)

**Pull request merged**